lib/optimizely/event_builder.rb

```diff
@@ -41,198 +42,202 @@ def ==(event)
+      # experiment_variation_map - +Hash+ Map of experiment IDs to bucketed variation IDs
+      #
+      # Returns +Hash+ Impression event params
 
-      attributes.keys.each do |attribute_key|
-        # Omit falsy attribute values
-        attribute_value = attributes[attribute_key]
-        next unless attribute_value
+      conversionEventParams = []
 
-        # Skip attributes not in the datafile
-        attribute_id = @config.get_attribute_id(attribute_key)
-        next unless attribute_id
+      experiment_variation_map.each do |experiment_id, variation_id|
 
-        feature = {
-          'id' => attribute_id,
-          'name' => attribute_key,
-          'type' => 'custom',
-          'value' => attribute_value,
-          'shouldIndex' => true,
+        decision = {
+          decisions: [{
+            campaign_id: @config.experiment_id_map[experiment_id]['layerId'],
+            experiment_id: experiment_id,
+            variation_id: variation_id,
+          }],
+          events: [],
         }
-        @params['userFeatures'].push(feature)
-      end
-    end
-
-    def add_decision(experiment, variation_id)
-      experiment_key = experiment['key']
-      experiment_id = experiment['id']
-      @params['layerId'] = @config.experiment_key_map[experiment_key]['layerId']
-      @params['decision'] = {
-        'variationId' => variation_id,
-        'experimentId' => experiment_id,
-        'isLayerHoldback' => false,
-      }
-    end
 
-    def add_event_tags(event_tags)
-      @params['eventFeatures'] ||= []
-      @params['eventMetrics'] ||= []
+        event_object = {
+          entity_id: @config.event_key_map[event_key]['id'],
+          timestamp: get_timestamp(),
+          uuid: get_uuid(),
+          key: event_key,
+        }
 
-      return if event_tags.nil?
+        if event_tags
+          event_revenue = Helpers::EventTagUtils.get_revenue_value(event_tags)
+          if event_revenue
+            event_object[:revenue] = event_revenue
+          end
 
-      event_tags.each_pair do |event_tag_key, event_tag_value|
-        next if event_tag_value.nil?
+          event_value = Helpers::EventTagUtils.get_event_value(event_tags)
+          if event_value
+            event_object[:value] = event_value
+          end
 
-        event_feature = {
-          'name' => event_tag_key,
-          'type' => 'custom',
-          'value' => event_tag_value,
-          'shouldIndex' => false,
-        }
-        @params['eventFeatures'].push(event_feature)
+          event_object[:tags] = event_tags
+        end
 
-      end
-
-      event_value = Helpers::EventTagUtils.get_revenue_value(event_tags)
+        decision[:events] = [event_object]
 
-      if event_value
-        event_metric = {
-          'name' => 'revenue',
-          'value' => event_value
-        }
-        @params['eventMetrics'].push(event_metric)
+        conversionEventParams.push(decision)
       end
-
+    
+      return conversionEventParams
```

Comment:
I notice that 'return' statement isn't used in other methods. Ruby auto returns the value of the last statement. I think Optimizely would prefer to have single convention followed

Reply:
done